File: backend/Models/TodoTask.cs
namespace backend.Models;

public class TodoTask
{
    public int Id { get; set; }

    public required string Title { get; set; }

    public string? Description { get; set; }

    public bool IsCompleted { get; private set; }

    public DateTime CreatedAt { get; private set; } = DateTime.UtcNow;

    public DateTime? DueDate { get; set; }

    public DateTime? CompletedAt { get; private set; }

    public TaskPriority Priority { get; set; } = TaskPriority.Medium;

    public int UserId { get; set; }

    public User User { get; set; } = null!;

    public void Complete()
    {
        if (IsCompleted)
            return;

        IsCompleted = true;
        CompletedAt = DateTime.UtcNow;
    }

    public void Reopen()
    {
        IsCompleted = false;
        CompletedAt = null;
    }
}

File: frontend/src/app/models/todo-task.ts
import { TaskPriority } from './task-priority';

export interface TodoTask {
  title: string;
  description: string;
  isCompleted: boolean;
  createdAt: string;
  completedAt: string | null;
  priority: TaskPriority;
  categoryId: number | null;
  categoryName: string | null;
}
